src/scripts/main.js

```diff
@@ -1 +1,68 @@
 'use strict';
+
+const firstRejectionMessage = `First promise was rejected`;
+const firstResolvedMessage = `First promise was resolved`;
+const secondResolvedMessage = `Second promise was resolved`;
+const thirdResolvedMessage = `Third promise was resolved`;
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (i) => {
+    if (i.button === 0) {
+      resolve(firstResolvedMessage);
+    }
+  });
+
+  setTimeout(() => {
+    reject(firstRejectionMessage);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (i) => {
+    if (i.button === 0 || i.button === 2) {
+      resolve(secondResolvedMessage);
+    }
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let leftClicked = false;
+  let rightClicked = false;
+
+  document.addEventListener('mousedown', (i) => {
+    if (i.button === 0) {
+      leftClicked = true;
+    }
+
+    if (i.button === 2) {
+      rightClicked = true;
+    }
+
+    if (leftClicked && rightClicked) {
+      resolve(thirdResolvedMessage);
+    }
+  });
```

Comment:
The third promise resolves when both left and right mouse buttons are clicked. However, if the user clicks both buttons simultaneously, the promise will resolve immediately. Consider whether this behavior is intended.